Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -179,6 +180,24 @@ public IEnumerable<Product> Fragment(DissociationType dissociationType, Fragment
 
             var productCollection = TerminusSpecificProductTypes.ProductIonTypesFromSpecifiedTerminus[fragmentationTerminus].Intersect(DissociationTypeCollection.ProductsFromDissociationType[dissociationType]);
 
+            List<(ProductType, int)> skippers = new List<(ProductType, int)>();
```

Comment:
I benchmarked this with a human DB, digesting each protein and fragmenting each peptide and found that there was negligible impact on performance (~30.5 seconds with or without this "skippers" code), so this appears to be fine

Reply:
I sort of botched the first benchmark because I used HCD. I benchmarked again with ETD and with skipping it takes about 38s, without skipping it takes 35s. Seems fine to me